### PR TITLE
Get tests running & give pseudocode in tests like Java assessment

### PR DESCRIPTION
**Pull request by twinlensreflex**
## What?
1. Updated sbt versions for Apple Silicon, updated Play version
2. Add pseudocode to tests showing desired assertions

## Why?
1. So that scala tests & play code run
2. The scala test set up was a little complex to get your head around so we copied the pseudocode assertion from the Java test cases as it is a little less confusing

### CODE REVIEW

File: election-api-scala/app/model/PartyResult.scala [hunk 1]
```diff
@@ -5,5 +5,7 @@ case class ConstituencyResult(id: Int, name: String, seqNo: Int, partyResults: S
 
 case class ApiResponse (error: String, message: String)
 
-
-case class Scoreboard(declared:Int)
\ No newline at end of file
+// TODO: this class should hold:
+//  - the overall winner (if there is one)
+//  - the seats that each party wins in Parliament
+case class Scoreboard(winner: String)
```
Comment (timsweetman) [inline election-api-scala/app/model/PartyResult.scala:11]: Do you want *winner* parameter here? Feels like prejudicing the approach taken...

File: election-api-scala/tasks.md [hunk 2]
```diff
@@ -16,16 +16,10 @@ The scoreboard should show:
...
 
 Bonus information for the scoreboard:
 - The total votes for each party
 - The total share of the vote for each party. So the percentage of votes for each party.
-
-
-### Possible other implementations
-
-- Absolute majority required. Someone needs 50% + 1 votes or a run off is triggered (check the data that's probably all constituencies)
-- Allocate the seats from the total declarations based on % of vote share
```
Comment (timsweetman) [inline election-api-scala/tasks.md:31]: Agree, the whole "other implementations" section looking at other voting systems is a bad idea (but! for fairness/consistency we should edit tasks.md for other languages)

File: election-api-scala/test/controllers/ResultsControllerSpec.scala [hunk 3]
```diff
@@ -21,47 +19,37 @@ class ResultsControllerSpec extends PlaySpec with GuiceOneAppPerTest with Inject
...
-      lab mustEqual LabSeats
-      con mustEqual ConSeats
-//      something mustEqual winner
-      fail("Implement me")
+      scoreboard must not be empty
+      // LD == 62
+      // LAB == 349
+      // CON == 210
+      // winner = LAB
     }
```
Comment (timsweetman) [inline election-api-scala/test/controllers/ResultsControllerSpec.scala:53]: 👍 on having comments stating assertions needed & letting the candidate implement (as in other languages)

File: election-api-scala/README.md [hunk 4]
```diff
@@ -21,7 +21,7 @@ The API has 3 endpoints:
 
 During your assessment we will ask you to work though the task in `tasks.md` with a pair. Please do not work on or complete these prior to the assessment.
 
-:warning:  If you make any changes to the code, please ensure you return it to it's initial (HEAD) state before your assessment.
+__Warning:__  If you make any changes to the code, please ensure you return it to it's initial (HEAD) state before your assessment.
```
Comment (timsweetman) [inline election-api-scala/README.md:24]: > it's initial state

Its. Not it's. Oh dear.

**Comment by timsweetman** [comment]
In my opinion:

* one PR for the scala-specific edits, which all look fine to me
* another for tasks edits which should span all languages

**Comment by timsweetman** [comment]
If this were just the code changes, not the prose changes to tasks.md, I'd merge!

File: election-api-scala/app/model/PartyResult.scala (hunk 1 above)
Comment (Willdotwhite) [reply election-api-scala/app/model/PartyResult.scala:11]: I'd like that - the other languages don't lead the code approach like this, it's just the tests that imply a `winner` field needs to be grabbed somehow.

**Pull request merged**